packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1430,22 +1430,97 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction status associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param options - The custodial transaction options to update.
+   * @param options.custodyStatus - The new custody status value to be assigned.
+   * @param options.errorMessage - The error message to be assigned in case transaction status update to failed.
+   * @param options.hash - The new hash value to be assigned.
+   * @param options.status - The new status value to be assigned.
+   */
+  updateCustodialTransaction(
+    transactionId: string,
+    {
+      custodyStatus,
+      errorMessage,
+      hash,
+      status,
+    }: {
+      custodyStatus?: string;
+      errorMessage?: string;
+      hash?: string;
+      status?: TransactionStatus;
```

Comment:
Should we define the explicit subset of status values that are supported here?

And also validate in the method that it's a valid status?

Reply:
I pushed some changes to validate the status. https://github.com/MetaMask/core/pull/2018/commits/fbef8f23fcfc26a68fecf4c2cafb8f92139052b8